### PR TITLE
[v6r12] Recursive LFNs owner/group/mode changes

### PR DESCRIPTION
**Pull request by calancha**
Recursion wasnt implemented on _changePathFunction (argument recursive was ignored).
This commit and PR #2632 hopefully make possible to call: chown, chgrp and chmod
passing the argument '-R' in the file catalog client.

### CODE REVIEW

**Comment by chaen** [comment]
Hi,

concerning this PR as well as this one https://github.com/DIRACGrid/DIRAC/pull/2632

A proper recursive behavior has been implemented for v6r15 which is to be the next release. Rather than submitting patch (very much appreciated btw :-) ) to this very old version, could you rather consider upgrading to a more recent one ? 

Cheers, and thanks

**Comment by andresailer** [comment]
Hi Chris,
Tino is one of my ILCDIRAC power-users, so no we can't update (InputDataResolution, xroot storage). :(
Sorry

**Comment by calancha** [comment]
Hi, thank you for the information.
I am glad to hear that news. As Andre said i am just an user so i cannot include anything myself.
Hopefully i will enjoy such new version in the future.

**Comment by chaen** [comment]
I see... then I let @atsareg look at it, but I would very strongly suggest to run the unit/integration tests on them before merging. Also, if it gets merged, I would avoid to propagate it to further versions.

File: DataManagementSystem/DB/FileCatalogComponents/FileManagerBase.py [hunk 1]
```diff
@@ -1401,6 +1401,15 @@ def _changePathFunction( self, paths, credDict, change_function_directory, chang
     for p in result['Value']['Successful']:
       if result['Value']['Successful'][p]:
         dirList.append( p )
+        if recursive:
+           result = self.db.dtree.getSubdirectories(p)
```
Comment (andresailer) [inline DataManagementSystem/DB/FileCatalogComponents/FileManagerBase.py]: Is this a three space indent?

Comment (calancha) [inline DataManagementSystem/DB/FileCatalogComponents/FileManagerBase.py]: Upps, yes, i used 3 spaces instead of 2 in that if construct. Sorry for that.

**Comment by andresailer** [comment]
I just tested this change, but it doesn't work. 
I would propose to not merge this and wait for the later version.

**Comment by calancha** [comment]
Ok, thank you for having tested.

**Pull request closed without merging**